develop/interoperability/xcm-runtime-apis.md

```diff
@@ -0,0 +1,360 @@
+
+    ***Usage with PAPI***
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-xcm-weight.js'
+    ```
+
+    !!!note
+        Make sure to replace `INSERT_USER_ADDRESS` with your SS58 address before running the script.
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-xcm-weight-output.html'
+
+    ---
+
+### Query Weight to Asset Fee
+
+Converts a given weight into the corresponding fee for a specified `AssetId`. It allows clients to determine the cost of execution in terms of the desired asset.
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/heads/stable2412/polkadot/xcm/xcm-runtime-apis/src/fees.rs:58:58'
+```
+
+??? interface "Input parameters"
+
+    `weight` ++"Weight"++ <span class="required" markdown>++"required"++</span>
+    
+    The execution weight to be converted into a fee.
+
+    ---
+
+    `asset` ++"VersionedAssetId"++ <span class="required" markdown>++"required"++</span>
+    
+    The asset in which the fee will be calculated. This must be a versioned asset ID compatible with the runtime.
+
+    ---
+
+??? interface "Output parameters"
+
+    ++"Result<u128, Error>"++
+    
+    The fee needed to pay for the execution for the given `AssetId.`
+
+    ---
+
+??? interface "Example"
+
+    This example demonstrates how to calculate the fee for a given execution weight using a specific versioned asset ID (PAS token) on Paseo Asset Hub.
+
+    ***Usage with PAPI***
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-weight-to-fee.js'
+    ```
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-weight-to-fee-output.html'
+    ---
+
+### Query Delivery Fees
+
+Retrieves the delivery fees for sending a specific XCM message to a designated destination. The fees are always returned in a specific asset defined by the destination chain.
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/heads/stable2412/polkadot/xcm/xcm-runtime-apis/src/fees.rs:68:68'
+```
+
+??? interface "Input parameters"
+
+    `destination` ++"VersionedLocation"++ <span class="required" markdown>++"required"++</span>
+    
+    The target location where the message will be sent. Fees may vary depending on the destination, as different destinations often have unique fee structures and sender mechanisms.
+
+    ---
+
+    `message` ++"VersionedXcm<()>"++ <span class="required" markdown>++"required"++</span>
+    
+    The XCM message to be sent. The delivery fees are calculated based on the message's content and size, which can influence the cost.
+
+    ---
+
+??? interface "Output parameters"
+
+    ++"Result<VersionedAssets, Error>"++
+    
+    The calculated delivery fees expressed in a specific asset supported by the destination chain. If an error occurs during the query, it returns an error instead.
+
+    ---
+
+??? interface "Example"
```

Comment:
Again, this setup with "input", "output" and "example" is useful, but I think the most useful thing we could do is show how these functions work together.

I would show:
- the whole process of how to estimate XCM fees
- how the dry run api can be used in an app to let the user know if an action will be successful or not

Reply:
This will be done in the tutorial